File: delete_col_sorted.cpp
#include "leetcode.h"

class Solution {
public:
  int minDeletionSize(vector<string> &strs) {
    int ans = 0;
    if (strs.empty())
      return ans;
    for (auto i = 0; i < strs[0].size(); ++i)
      for (auto j = 1; j < strs.size(); ++j)
        if (strs[j - 1][i] > strs[j][i]) {
          ++ans;
          break;
        }
    return ans;
  }
};

int main() {
  Solution obj;
  vector<string> strs1 = {"cba", "daf", "ghi"}, strs2 = {"a", "b"},
                 strs3 = {"zyx", "wvu", "tsr"};
  cout << obj.minDeletionSize(strs1) << endl;
  cout << obj.minDeletionSize(strs2) << endl;
  cout << obj.minDeletionSize(strs3) << endl;
}

File: delete_col_sorted.rs
struct Solution;

impl Solution {
    pub fn min_deletion_size(strs: Vec<String>) -> i32 {
        let mut ans = 0;
        let mut new_strs: Vec<Vec<char>> = Vec::new();
        for i in 0..strs.len() {
            new_strs.push(strs[i].chars().collect::<Vec<char>>());
        }
        for i in 0..new_strs[0].len() {
            for j in 1..new_strs.len() {
                if new_strs[j - 1][i] > new_strs[j][i] {
                    ans += 1;
                    break
                }
            }
        }
        ans
    }
}

fn main() {
    let strs = vec![String::from("cba"), String::from("daf"), String::from("ghi")];
    print!("{}", Solution::min_deletion_size(strs));
}
